pkg/controller/appdefinition/generation.go

```diff
@@ -10,14 +10,11 @@ func UpdateObservedFields(req router.Request, resp router.Response) error {
 	app := req.Object.(*v1.AppInstance)
 	app.Status.ObservedImageDigest = app.Status.AppImage.Digest
 	app.Status.ObservedGeneration = app.Generation
-	app.Status.ObservedAutoUpgrade = impliedAutoUpgrade(app.Spec)
+	app.Status.ObservedAutoUpgrade = autoUpgradeEnabled(app.Spec)
```

Comment:
Is this field still needed?

Reply:
Yes, I believe so. Namely for the case that a pattern is used but autoupgrade is not set. More context [here](https://github.com/acorn-io/runtime/pull/2002#discussion_r1281080837).